x-pack/legacy/plugins/watcher/public/services/watches/watches.js

```diff
@@ -32,10 +28,9 @@ export class WatchesService {
     // $http.delete does not take the request body as the 2nd argument. Instead it expects the 2nd
     // argument to be a request options object, one of which can be the request body (data). We also
     // need to explicitly define the content type of the data.
-    const body = {
-      watchIds
-    };
-    return this.$http.post(`${this.basePath}/watches/delete`, body)
-      .then(response => response.data.results);
+    return http.post(`${ROUTES.API_ROOT}/watches/delete`, {
+      body: JSON.stringify({ watchIds })
+    })
+      .then(response => response.results);
```

Comment:
I guess since it is not Angular anymore, we can use `async/await`.

```js
  async deleteWatches(watchIds) {
    const {result} = await http.post(`${ROUTES.API_ROOT}/watches/delete`, {
      body: JSON.stringify({ watchIds })
    });

    return result;
  }
```

Reply:
Do you know how that would work with the angular digest cycle? 

Is it better to return a value or a promise for an async call? I'm leaning toward promise as there's more functionality and yet its easy to `await` if you just want the value.

Reply:
Making the function `async` will [already implicitly return a promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) that resolves with the returned value, so you would still be returning a promise with @streamich's example above.